library/core/qsl_base/src/main/java/org/quiltmc/qsl/base/api/util/QuiltUtil.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.base.api.util;
+
+import java.util.function.Function;
+
+/**
+ * A set of various utilities.
+ */
+public final class QuiltUtil {
```

Comment:
I'm not sure why this needs to be public api.

Reply:
Well, I could put the method in QuiltBaseImpl instead but it would still be public, so I thought why not just generalize it and make it available instead?

Reply:
Okay if this is going to be public, we should probably name it more appropriately, something like `QuiltAssertions`?

Reply:
If we don't intend to have more utilities methods that are not assertions in this class, then yes, would be a good idea.